runtime/complex.go

```diff
@@ -140,3 +144,91 @@ func complexHash(f *Frame, o *Object) (*Object, *BaseException) {
 	}
 	return NewInt(hashCombined).ToObject(), nil
 }
+
+func complexArithmeticOpComplex(f *Frame, method string, v, w complex128, fun func(v, w complex128) complex128) (*Object, *BaseException) {
+	return NewComplex(fun(v, w)).ToObject(), nil
+}
+
+func complexArithmeticOp(f *Frame, method string, v complex128, w float64, fun func(v complex128, w float64) complex128) (*Object, *BaseException) {
```

Comment:
Seems like you could fold all the coercion and overflow checking logic into this function so you don't have to duplicate it for every operator.